Fix bugs

soywod opened the pull request.

soywod committed:
fix signatures + tests

soywod commented on src/config/model.rs:
````diff
@@ -110,59 +110,6 @@ impl Account {
-    ///     assert_eq!("BestEmail@Ever.lol".to_string(), address);
-    /// }
-    /// ```
-    pub fn get_full_address(&self) -> String {
````
I removed this function because it already exists (see `Config::address`). As mentioned previously, an address should be fetched at the config level, because accounts "inherit" properties from config.

TornaxO7 replied:
Yes, I understood now :+1:

soywod commented on src/config/model.rs:
```diff
@@ -110,59 +110,6 @@ impl Account {
-        if let Some(name) = &self.name {
-            format!("{} <{}>", name, self.email)
-        } else {
-            format!("{}", self.email)
```
This case should never happen, because the name is a mandatory option at the config level. That's why it's not present in `Config::address`.

TornaxO7 replied:
Aye

soywod commented on src/msg/model.rs:
```diff
@@ -928,92 +930,130 @@ impl fmt::Display for Msgs {
+        let output = Output::default();
+        let arg_matches = ArgMatches::default();
+
+        let ctx = Ctx::new(&config, &account, &output, "INBOX", &arg_matches);
```
I agree that there is a bit too much boilerplate (mostly because of the config). I will find a way to simplify it (I was thinking about writing a testing macro).

soywod commented:
I notice that there is a lot of indentations / new lines difference between my code and yours. Did you properly activate your formater (the standard one, rustfmt)?

soywod committed:
fix body and signature new lines

TornaxO7 commented:
Damn, I thought that if I click on `ready for review` that it just brings me to the diff to see the diff, but it converts this PR from a draft to a complete PR instead. Sorry!

TornaxO7 commented on src/config/model.rs:
```diff
@@ -415,7 +365,7 @@ impl Config {
             .and_then(|sig| fs::read_to_string(sig).ok())
             .or_else(|| sig.map(|sig| sig.to_owned()))
-            .map(|sig| sig_delim.to_owned() + sig.as_ref())
+            .map(|sig| String::from("\n") + sig_delim + sig.as_ref())
```
Hm... can this be changed to `format!("\n{}{}", sig_delim, sig)`?

soywod replied:
Yes I agree, I will push the changes soon :heavy_check_mark:

TornaxO7 commented on src/msg/cli.rs:
```diff
@@ -444,11 +444,7 @@ fn msg_matches_reply(ctx: &Ctx, matches: &clap::ArgMatches) -> Result<bool> {
-    } else {
-        msg.change_to_reply(&ctx.account, false)?;
-    }
+        msg.change_to_reply(&ctx, matches.is_present("reply-all"))?;
```
Good point! :D

TornaxO7 commented on src/msg/model.rs:
```diff
@@ -928,92 +929,130 @@ impl fmt::Display for Msgs {
+        let msg = Msg::new(&ctx);
+        let expected_envelope = Envelope {
+            from: vec![String::from("Config Name <test@mail.com>")],
+            signature: Some(String::from("\n-- \nAccount Signature")),
```
One question: Is the space after the two lines `-- ` really needed? What's the benefit of doing it like this?

soywod replied:
It's not about benefits, but more about standards, see https://en.wikipedia.org/wiki/Signature_block#Standard_delimiter. I also refer to Thunderbird (they seem to have included a lot of standards), and there is the space also.

TornaxO7 replied:
Wow, that's a "weird" standard in my opinion. Thanks for the link and the explanation!

soywod commented:
It's fine, just don't merge until I'm fully ready :laughing:

TornaxO7 commented:
I have a question: Wouldn't it make more sense to let `Ctx` store fully its attributes instead of references to them? Because if we're mainly using `Ctx` than we can discard the other variables as well.

TornaxO7 commented:
> I notice that there is a lot of indentations / new lines difference between my code and yours. Did you properly activate your formater (the standard one, rustfmt)?

Yes, I noticed that as well. I'm on rustup-nightly and this is my formatter config (~/.config/rustfmt/rustfmt.toml`):

```toml
color = "Always"
enum_discrim_align_threshold = 100
format_code_in_doc_comments = true
# format_strings = true
match_block_trailing_comma = false
# max_width = 80
# reorder_impl_items = true
struct_field_align_threshold = 200
wrap_comments = true
```

Should I use the default formatting settings?

soywod commented:
> Should I use the default formatting settings?

I would say yes, because I did not touch the rustfmt config. Is it possible to add a `rustfmt.toml` at the root of the project, so everyone working on Himalaya will have the same formatting?

TornaxO7 commented:
> I would say yes, because I did not touch the rustfmt config. Is it possible to add a rustfmt.toml at the root of the project, so everyone working on Himalaya will have the same formatting?

[Looks like it's possible](https://github.com/rust-lang/rustfmt#configuring-rustfmt)

TornaxO7 reviewed (commented):
A short and little pre-review.

soywod commented:
> [Looks like it's possible](https://github.com/rust-lang/rustfmt#configuring-rustfmt)

Awesome, I will add the standard config the project then.

soywod committed:
fix reply indentation and signature new lines

soywod committed:
add default rustfmt.toml

soywod committed:
apply fmt on all the project

soywod committed:
fix msg tests

soywod commented:
The branch is ready for merge. I fixed most of the bugs I saw. I also added the `rustfmt.toml` (I generated a default one). I will open another issue with all the questions / points I had in mind (to be treated after this PR).

TornaxO7 reviewed (approved):
Nice!
